lib/src/server.dart

```diff
@@ -170,22 +173,21 @@ class Server {
         'result': result,
         'id': request['id']
       };
-    }).catchError((error, stackTrace) {
-      if (error is! RpcException) {
-        error = new RpcException(
-            error_code.SERVER_ERROR, getErrorMessage(error), data: {
-          'full': error.toString(),
-          'stack': new Chain.forTrace(stackTrace).toString()
-        });
-      }
-
+    } catch (error, stackTrace) {
       if (error.code != error_code.INVALID_REQUEST &&
```

Comment:
Just pulling the `request.containsKey()` check would change the logic since it's part of an &&. Although, actually my change also changes the logic as the created RpcException could pass this check.

There doesn't seem to be any way to easily simplify this; returning in the `if (error is! RpcException` would result in quite a bit of duplication.